web-common/src/features/dashboards/stores/dashboard-stores.ts

```diff
@@ -451,10 +453,12 @@ const metricViewReducers = {
   setSelectedComparisonRange(
     name: string,
     comparisonTimeRange: DashboardTimeControls,
+    metricsViewSpec: V1MetricsViewSpec,
   ) {
     updateMetricsExplorerByName(name, (metricsExplorer) => {
       setDisplayComparison(metricsExplorer, true);
       metricsExplorer.selectedComparisonTimeRange = comparisonTimeRange;
+      AdvancedMeasureCorrector.correct(metricsExplorer, metricsViewSpec);
```

Comment:
**Issue**: The method 'AdvancedMeasureCorrector.correct' is called without any null checks on 'metricsExplorer' or 'metricsViewSpec'. This could lead to a runtime error if either is null. <br> **Fix**: Add null checks before calling the 'AdvancedMeasureCorrector.correct' method to ensure stability. <br> **Code Suggestion**: 
 ```
 @@ -460,3 +460,5 @@
      metricsExplorer.selectedComparisonTimeRange = comparisonTimeRange;
 +    if (metricsExplorer && metricsViewSpec) {
 +      AdvancedMeasureCorrector.correct(metricsExplorer, metricsViewSpec);
      }
 -    AdvancedMeasureCorrector.correct(metricsExplorer, metricsViewSpec);
 ```